configs/myconfigs/sampling/Sampling.py

```diff
@@ -0,0 +1,361 @@
+                      help="Simulated functional warmup time in ms")
+SimpleOpts.add_option("--detailed_warmup_time", default="10us",
+                      help="Simulated detailed warmup time in ms")
+SimpleOpts.add_option("--detailed_time", default="1ms",
+                      help="Simulated time in ms for detailed measurements, \
+                          (size of measurment window)")
+
+from sim_tools import *
+
+
+_valid_system = False
+sample_insts = []            #-- list storing instructions for sample points
+_sample = 0                  #-- current sample index
+
+
+def generateSamples(system, opts, instructions, samples, \
+                    start_insts):
+    """ Generates a fixed number of random sample points that can
+        be used by other methods of this module.
+
+        @param system the system we are running
+        @param opts command line options as an instance of optparse
+        @param instructions total instructions in the ROI (summed over all
+               CPUs)
+        @param samples number of sample points in the ROI
+        @param start_insts instructions executed till the start of roi
+    """
+    from random import randint, sample, seed
+    global sample_insts
+    seed()
+
+    warmup_time = opts.warmup_time
+    detailed_warmup_time = opts.detailed_warmup_time
+    detailed_time = opts.detailed_time
+
+    # We probably want to updated the max instructions by subtracting the
+    # insts that we are going to execute in the last detailed sim.
+    sample_secs = toLatency(warmup_time) + toLatency(detailed_warmup_time) \
+                    + toLatency(detailed_time)
+    # assume 1 IPC per CPU
+    instructions -= int(sample_secs * system.clk_domain.clock[0].frequency *
+                        len(system.cpu) * 1)
+
+    # Get the instruction numbers that we want to exit at for our sample
+    # points. It needs to be sorted, too.
+    sample_insts = sample(xrange(start_insts,
+                                 start_insts + instructions),
+                          samples)
+    sample_insts.sort()
+
+
+
+def dumpSamples(outfile="random.samples"):
+    """ Dumps the sample points in a file so that they can
+        be used accross multiple parallel jobs
+
+        @param outfile name of the sample dump file
+    """
+    import pickle
+    rfilename = "%s/%s" % (m5.options.outdir, outfile)
+
+    print "Dumping random sample points at :", rfilename
+    rfile = open(rfilename , "w")
+    pickle.dump(sample_insts, rfile)
+    rfile.close()
+
+
+
+def loadSamples(samplefile):
+    """ Loads sample points from a file dumped previously
+        using dumpSamples().
+
+        @param samplefile name of the sample dump file
+    """
+    import pickle
+    global sample_insts
+
+    print "Loading random samples : ", samplefile
+    if not os.path.isfile(samplefile):
+        print >> sys.stderr, "Error could not find file", samplefile
+        sys.exit(1)
+
+    rfile = open(samplefile, "r")
```

Comment:
Just for fun, it's safer to use `with open(samplefile, "r") as rfile` Then you don't need to close the file, either.